src/layers/models/layersManager.ts

```diff
@@ -43,79 +42,64 @@ class LayersManager {
+        const currentSourceCache: IMapProxyCache = configJson.caches[sourceName];
+        return (currentSourceCache.cache.type == cacheType)
+      });
 
-      if (!isLayerNameExists(jsonDocument, mosaicName)) {
-        throw new NotFoundError(`Mosaic name '${mosaicName}' is not exists`);
+      if (requestedCache != null) {
+        return { cacheName: requestedCache };
       }
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      const mosaicCache: IMapProxyCache = jsonDocument.caches[mosaicName];
-      mosaicCache.sources.push(layerToMosaicRequest.layerName);
-      return jsonDocument;
-    };
-
-    await this.configProvider.updateJson(editJson);
-    this.logger.info(`Successfully added layer: '${layerToMosaicRequest.layerName}' to mosaic: '${mosaicName}'`);
+    }
+    throw new NotFoundError(errorMsg)
   }
 
-  public async updateMosaic(mosaicName: string, updateMosaicRequest: IUpdateMosaicRequest): Promise<void> {
-    this.logger.info(`Update mosaic: ${mosaicName} request`);
-
+  public async addLayer(layerRequest: ILayerPostRequest): Promise<void> {
     const editJson = (jsonDocument: IMapProxyJsonDocument): IMapProxyJsonDocument => {
-      if (!isLayerNameExists(jsonDocument, mosaicName)) {
-        throw new NotFoundError(`Mosaic name '${mosaicName}' is not exists`);
+      if (isRedisCacheLayer(layerRequest.name)) {
+        const errorMsg = `layer names that ends with '-redis' are not supported`;
+        throw new NotImplementedError(errorMsg);
       }
-      updateMosaicRequest.layers.forEach((layer) => {
-        if (!isLayerNameExists(jsonDocument, layer.layerName)) {
-          throw new NotFoundError(`Layer name '${layer.layerName}' is not exists`);
-        }
-      });
 
-      const sortedLayers: string[] = sortArrayByZIndex(updateMosaicRequest.layers);
+      this.addNewCache(jsonDocument, layerRequest);
 
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      const mosaicCache: IMapProxyCache = jsonDocument.caches[mosaicName];
-      mosaicCache.sources = sortedLayers;
       return jsonDocument;
     };
 
     await this.configProvider.updateJson(editJson);
-    this.logger.info(`Successfully updated mosaic: '${mosaicName}'`);
   }
 
-  public getAllLinkedCaches(baseCacheNames: string[], mapproxyConfiguration: IMapProxyJsonDocument): string[] {
+  public getAllLayerLinkedCaches(baseCacheNames: string[], mapproxyConfiguration: IMapProxyJsonDocument): string[] {
     const linkedCaches: string[] = [];
     const baseCacheNamesDuplicate: string[] = [...baseCacheNames];
 
     baseCacheNamesDuplicate.forEach((currentCache) => {
-      const sourceName = currentCache.endsWith('-source') ? currentCache : `${currentCache}-source`;
-      const redisSourceName = currentCache.endsWith('-source') ? currentCache.replace('-source', '') : currentCache;
+      const mainCacheName = isRedisCacheLayer(currentCache) ? currentCache.replace('-redis', '') : currentCache;
```

Comment:
all '-redis' methods should be in utils same as in here -
https://github.com/MapColonies/mapproxy-api/pull/126/files#diff-a404995816c64ab594401ce132110c5ea7926a0e27d6bc19cd0a59a866ece081R45

Reply:
done